src/test/java/net/minecraftforge/test/TestCapabilityMod.java

```diff
@@ -58,13 +59,28 @@ public void onInteract(PlayerInteractEvent.LeftClickBlock event)
             IExampleCapability inv = te.getCapability(TEST_CAP, event.getFace());
             System.out.println("Hi I'm a " + inv.getOwnerType());
         }
-        if (event.getWorld().getBlockState(event.getPos()).getBlock() == Blocks.DIRT)
+        if (event.getWorld().getBlockState(event.getPos()).getBlock() == Blocks.DIRT && event.getItemStack().hasCapability(TEST_CAP, null))
         {
-            event.getEntityPlayer().addChatMessage(new TextComponentString(TextFormatting.RED + "" + TextFormatting.ITALIC + "TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST"));
+            IExampleCapability cap = event.getItemStack().getCapability(TEST_CAP, null);
+            event.getEntityPlayer().addChatMessage(new TextComponentString((cap.getVal() ? TextFormatting.GREEN : TextFormatting.RED) + "" + TextFormatting.ITALIC + "TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST TEST"));
             event.setCanceled(true);
         }
     }
 
+    @SubscribeEvent
+    public void onInteractItem(PlayerInteractEvent.RightClickItem event)
+    {
+        if (!event.getEntityPlayer().isSneaking()) return;
+        if (event.getItemStack().func_190926_b()) return;
```

Comment:
Is this `isEmpty()`? This is not that necessary because of the check in the line below.